go/database/mpt/archive_trie_test.go

```diff
@@ -1690,6 +1684,105 @@ func TestArchiveTrie_FailingLiveStateUpdate_InvalidatesArchive(t *testing.T) {
 	}
 }
 
+func TestArchiveTrie_VisitTrie_CorrectDataIsVisited(t *testing.T) {
+	for _, config := range allMptConfigs {
+		t.Run(config.Name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			archive, err := OpenArchiveTrie(t.TempDir(), config, NodeCacheConfig{Capacity: 1024})
```

Comment:
This test could be also implemented using TableDriven testing: https://go.dev/wiki/TableDrivenTests
where the subject of customisation would be `NodeVisitor` because at the moment the test contains two blocks of code that are the same, just the mock is different.

Reply:
Good point. I will rework the test.